File: src/main/java/com/digsol/turnomed/model/Address.java
package com.digsol.turnomed.model;

import com.digsol.turnomed.dto.AddressDTO;
import com.digsol.turnomed.dto.DtoCompatible;
import com.google.gson.JsonParser;
import lombok.Data;
import lombok.ToString;
import lombok.extern.slf4j.Slf4j;


import java.util.List;

@Data
@Slf4j
public class Address implements DtoCompatible<AddressDTO> {

    private Integer id;

    private String city;

    private String name;

    private String address;

    private List<Integer> sportTypes;

    private String info;

    @Override
    public AddressDTO toDto() {
        AddressDTO addressDTO = new AddressDTO();
        addressDTO.setId(this.getId());
        addressDTO.setCity(this.getCity());
        addressDTO.setClinicName(this.getName());
        addressDTO.setAddress(this.address);
        addressDTO.setDoctorTypes(this.sportTypes);
        addressDTO.setInfo(this.info != null ? new JsonParser().parse(this.info).getAsJsonObject() : null);
        return addressDTO;
    }
}

File: src/main/kotlin/com/digsol/turnomed/model/Address.kt
package com.digsol.turnomed.model

import com.digsol.turnomed.dto.AddressDTO
import com.digsol.turnomed.dto.DtoCompatible
import com.google.gson.JsonParser
import lombok.extern.slf4j.Slf4j

@Slf4j
data class Address(
    var id: Integer? = null,
    var city: String? = null,
    var name: String? = null,
    var address: String? = null,
    var sportTypes: List<Integer>? = null,
    var info: String? = null
) : DtoCompatible<AddressDTO> {

    override fun toDto(): AddressDTO {
        val addressDTO = AddressDTO()
        addressDTO.id = this.id
        addressDTO.city = this.city
        addressDTO.clinicName = this.name
        addressDTO.address = this.address
        addressDTO.doctorTypes = this.sportTypes
        addressDTO.info = this.info?.let { JsonParser().parse(it).asJsonObject }
        return addressDTO
    }
}
